docs/architecture-overview.rst

```diff
@@ -0,0 +1,311 @@
+
+We maintain several separate repositories for integrations and other
+glue code: a `Hubot adapter <https://github.com/zulip/hubot-zulip>`__
+; integrations with `Phabricator
+<https://github.com/zulip/phabricator-to-zulip>`__, `Jenkins
+<https://github.com/zulip/zulip-jenkins-plugin>`__, `Puppet
+<https://github.com/matthewbarr/puppet-zulip>`__, `Redmine
+<https://github.com/zulip/zulip-redmine-plugin>`__, and `Trello
+<https://github.com/zulip/trello-to-zulip>`__); `node.js API bindings
+<https://github.com/zulip/zulip-node>`__; and our `full-text search
+PostgreSQL extension <https://github.com/zulip/tsearch_extras>`__ .
+
+Our mobile clients are separate code repositories: `Android
+<https://github.com/zulip/zulip-android>`__, `iOS (stable)
+<https://github.com/zulip/zulip-ios>`__ , and `our experimental React
+Native iOS app <https://github.com/zulip/zulip-mobile>`__. Our `desktop
+application <https://github.com/zulip/zulip-desktop>`__ is also a
+separate repository.
+
+We use `Transifex <https://www.transifex.com/zulip/zulip/>`__ to do
+translations.
+
+In this overview we'll mainly discuss the core Zulip server and
+web application.
+
+
+Usage assumptions and concepts
+==============================
+
+Zulip is a real-time web-based chat application meant for companies
+and similar groups ranging in size from a small team to more than a
+thousand users. It features real-time notifications, message
+persistence and search, public group conversations (*streams*),
+invite-only streams, private one-on-one and group conversations,
+inline image previews, team presence/a buddy list, a rich API, and
+several integrations with other services. The maintainer team aims to
+support users who connect to Zulip using dedicated iOS, Android,
+Linux, Windows, and Mac OS X clients, as well as people using modern
+web browsers or dedicated Zulip API clients.
+
+A server can host multiple Zulip *realms* (organizations) at the same
+domain, each of which is a private chamber with its own users,
+streams, customizations, and so on. This means that one person might
+be a user of multiple Zulip realms. The administrators of a realm can
+choose whether to allow anyone to register an account and join, or
+only allow people who have been invited, or restrict registrations to
+members of particular groups (using email domain names or corporate
+single-sign-on login for verification). For more on scalability and
+security considerations, see `Zulip in production
+<https://github.com/zulip/zulip/blob/master/README.prod.md>`__.
+
+The default Zulip home screen is like a chronologically ordered inbox;
+it displays messages, starting at the oldest message that the user
+hasn't viewed yet. The home screen displays the most recent messages
+in all the streams a user has joined (except for the streams they've
+muted), as well as private messages from other users, in strict
+chronological order. A user can "narrow" to view only the messages in
```

Comment:
Italicize instead of quoting `narrow` to match the style of how you introduce other zulip terminology?